firefox_puppeteer/tests/test_utils.py

```diff
@@ -0,0 +1,49 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from firefox_ui_harness import FirefoxTestCase
+
+
+class TestSanitize(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Clear all previous history and cookies.
+        self.places.remove_all_history()
+        self.marionette.delete_all_cookies()
+
+        urls = [
+            'layout/mozilla_projects.html',
+            'layout/mozilla.html',
+            'layout/mozilla_mission.html',
+            'cookies/cookie_single.html'
+        ]
+        urls = [self.marionette.absolute_url(t) for t in urls]
+
+        # Open the test urls, including the single cookie setting page.
+        def load_urls():
+            with self.marionette.using_context('content'):
+                for url in urls:
+                    self.marionette.navigate(url)
+        self.places.wait_for_visited(urls, load_urls)
+        self.assertEqual(self.places.get_all_urls_in_history(), urls)
+        self.assertIsNotNone(self.marionette.get_cookie('litmus_1'))
+
+    def tearDown(self):
+        try:
+            self.places.remove_all_history()
+            self.marionette.delete_all_cookies()
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_sanitize_history(self):
+        """ Clears history. """
+        self.utils.sanitize(data_type={"history": True})
+        self.assertEqual(len(self.places.get_all_urls_in_history()), 0)
```

Comment:
Lets not compare the length but the list of urls the places method returns with an empty array. If clearing the history fails we can at least see which urls are still not removed.